client/lib/signup/step-actions/index.js

```diff
@@ -296,15 +297,16 @@ export function createSiteWithCart( callback, dependencies, stepData, reduxStore
 			domainItem,
 			themeItem,
 		};
-		processItemCart(
+		processItemCart( {
```

Comment:
As I understand it, we only want to effect this `createSiteWithCart` case of calling `processItemCart`. Therefore I think that this is a better place to add the hacky logic for our flow, and then you could just pass in the `skipSitesAndUsersFetch` flag.